webapp/src/Controller/Team/SubmissionController.php

```diff
@@ -201,4 +207,45 @@ public function viewAction(Request $request, int $submitId)
             return $this->render('team/submission.html.twig', $data);
         }
     }
+
+    /**
+     * @Route("/submission/{submitId<\d+>}/download", name="team_submission_download")
+     * @param int $submitId
+     *
+     * @return Response
+     * @throws NonUniqueResultException
+     */
+    public function downloadAction($submitId)
+    {
+        $allowDownload = (bool)$this->config->get('allow_team_submission_download');
+        if (!$allowDownload) {
+            throw new NotFoundHttpException('Submission download not allowed');
+        }
+
+        $user = $this->dj->getUser();
+        $team = $user->getTeam();
+        /** @var Submission $submission */
+        $submission = $this->em->createQueryBuilder()
+            ->from(Submission::class, 's')
+            ->join('s.judgings', 'j')
+            ->join('s.contest_problem', 'cp')
+            ->join('cp.problem', 'p')
+            ->join('s.language', 'l')
+            ->join('s.files', 'f')
+            ->select('s, f')
+            ->andWhere('j.submitid = :submitId')
+            ->andWhere('j.valid = 1')
+            ->andWhere('s.team = :team')
+            ->setParameter(':submitId', $submitId)
+            ->setParameter(':team', $team)
+            ->getQuery()
+            ->getOneOrNullResult();
```

Comment:
Why all these extra joins and conditions? I don't see the need to join on judgings, language or even problem and contest_problem.

Reply:
We don’t need them. Was a copy paste. Will remove it